packages/protocol/docs/Derivation.md

```diff
@@ -52,40 +52,63 @@ Throughout this document, metadata references follow the notation `metadata.fiel
+  bytes32 coreStateHash;
+  /// @notice Hash of the Derivation struct containing additional proposal data.
+  bytes32 derivationHash;
+}
+```
+
+The `Derivation` struct contains additional proposal data not needed during proving:
+
+```solidity
+/// @notice Contains derivation data for a proposal that is not needed during proving.
+/// @dev This data is hashed and stored in the Proposal struct to reduce calldata size.
+struct Derivation {
   /// @notice The L1 block number when the proposal was accepted.
   uint48 originBlockNumber;
-  /// @notice Whether the proposal is from a forced inclusion.
-  bool isForcedInclusion;
+  /// @notice The hash of the origin block.
+  bytes32 originBlockHash;
   /// @notice The percentage of base fee paid to coinbase.
   uint8 basefeeSharingPctg;
-  /// @notice Blobs that contains the proposal's manifest data.
+  /// @notice Array of derivation sources, where each can be regular or forced inclusion.
+  DerivationSource[] sources;
+}
+
+/// @notice Represents a source of derivation data within a Derivation
+struct DerivationSource {
+  /// @notice Whether this source is from a forced inclusion.
+  bool isForcedInclusion;
+  /// @notice Blobs that contain the source's manifest data.
   LibBlobs.BlobSlice blobSlice;
 }
 ```
 
-The following metadata fields are directly extracted from the `proposal` object:
+The following metadata fields are extracted from the `proposal` and `derivation` objects:
 
-| Metadata Field                | Value Assignment              |
-| ----------------------------- | ----------------------------- |
-| `metadata.id`                 | `proposal.id`                 |
-| `metadata.proposer`           | `proposal.proposer`           |
-| `metadata.timestamp`          | `proposal.timestamp`          |
-| `metadata.originBlockNumber`  | `proposal.originBlockNumber`  |
-| `metadata.basefeeSharingPctg` | `proposal.basefeeSharingPctg` |
-| `metadata.isForcedInclusion`  | `proposal.isForcedInclusion`  |
+| Metadata Field                | Value Assignment                    |
+| ----------------------------- | ----------------------------------- |
+| `metadata.id`                 | `proposal.id`                      |
+| `metadata.proposer`           | `proposal.proposer`                 |
+| `metadata.timestamp`          | `proposal.timestamp`                |
+| `metadata.originBlockNumber`  | `derivation.originBlockNumber`      |
+| `metadata.basefeeSharingPctg` | `derivation.basefeeSharingPctg`     |
+| `metadata.isForcedInclusion`  | `derivation.sources[i].isForcedInclusion` |
 
-The `blobSlice` field serves as the primary mechanism for locating and validating the proposal's manifest. The manifest structure is defined as follows:
+The `derivation.sources` array contains `DerivationSource` objects, each with a `blobSlice` field that serves as the primary mechanism for locating and validating the proposal's manifest. The manifest structure is defined as follows:
```

Comment:
As we mentioned below, each `BlobSlice` in `derivation.sources` will be decompressed + decoded into one `ProposalManifest`, so for each Shasta proposal, we might have several `ProposalManifest`s (with different `bytes proverAuthBytes` and `BlockManifest[] blocks`), but when assembling anchor transaction for each block within a multi-sources Shasta proposal, we still want the `_blockIndex` parameter to be monotonically increased across those several `ProposalManifest.blocks`, right?  Which might also mean that only the first `derivation.sources`'s `proverAuthBytes` will be used? Shall we also make this part clearer inside this doc?